### ISSUE TITLE
Cursor type 3 fails when first instrument(s) hidden

### ISSUE THREAD

**Issue by boblete** [opensheetmusicdisplay/opensheetmusicdisplay#1426]
Setting a cursor to type 3 and trying to get it to highlight when you have hidden the first instruments of the sheet fails.

The cursor no longer renders at all using the iterator presuming that the score is invisible. 

If the first Instrument on the sheet is visible then it highlights as expected:

SO I am hiding parts by setting vidibility:
```
 this.osmd.sheet?.Instruments?.forEach((e,i)=>{
            const lbl = e.nameLabel.text.toLowerCase();
            e.Visible = (
                lbl.indexOf(instrument.toLowerCase()) > -1 && 
                lbl.indexOf(partHeading.toLowerCase()) > -1
            );
        })
```
        
      if the first sheet is visible `this.osmd.sheet?.Instruments[0]` then it behaves as expected 
      
     Is there another way to hide sheets so the cursor stays active? otherwise this is a defect.

**Comment by sschmidTU** [comment]
You mean "another way to hide instruments" instead of "another way to hide sheets", right?
I'll have a look at the cursor not rendering.

**Comment by sschmidTU** [comment]
It works for me, though the area is slightly shifted to the right when the first instrument is not visible.

@boblete Can you give a sample and code to reproduce this in our demo?

**Comment by boblete** [comment]
Hi, I think i mean another way to hide instruments that wors with cursor 3 yes sorry

 -  yes I can sort a standalone demo out Monday hopefully - 

Reading the cursor code it looks like the code for the update cursor only reads the first returned measure for the calculations?

   const measurePositionAndShape: BoundingBox = this.graphic.findGraphicalMeasure(currentMeasureIndex, 0).PositionAndShape;
   
   THis is rendering a bounding box for the current position and shape of the first stave - the 0? if that stave is not visible then the calculations for the width in cursor 3 end up being 0? 

Should it be using the findVisibleGraphicalMeasure for the width or am I reading that wrong?

**Comment by sschmidTU** [comment]
You're probably right about the code. But I've used cursor type 3 and set the first instrument to Visible false and I didn't get an error (for a simple 2-instrument score). So that's why I'm asking for a way to reproduce

**Comment by sschmidTU** [comment]
This should fix it, right?

```diff
diff --git a/src/OpenSheetMusicDisplay/Cursor.ts b/src/OpenSheetMusicDisplay/Cursor.ts
index 14f969e4..a6d17a98 100644
--- a/src/OpenSheetMusicDisplay/Cursor.ts
+++ b/src/OpenSheetMusicDisplay/Cursor.ts
@@ -216,7 +216,14 @@ export class Cursor {
     height = endY - y;
 
     // Update the graphical cursor
-    const measurePositionAndShape: BoundingBox = this.graphic.findGraphicalMeasure(currentMeasureIndex, 0).PositionAndShape;
+    let visibleMeasure: GraphicalMeasure;
+    for (const gMeasure of this.graphic.MeasureList[currentMeasureIndex]) {
+      visibleMeasure = gMeasure;
+      if (gMeasure?.ParentStaffLine.ParentStaff.ParentInstrument.Visible) {
+        break;
+      }
+    }
+    const measurePositionAndShape: BoundingBox = visibleMeasure.PositionAndShape;
     this.updateWidthAndStyle(measurePositionAndShape, x, y, height);
 
     if (this.openSheetMusicDisplay.FollowCursor && this.cursorOptions.follow) {
```

```js
osmd.Sheet.Instruments[0].Visible = false;
osmd.cursor.cursorOptions.type = 3;
```

<img width="158" alt="image" src="https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/assets/33069673/6d4a8f15-9eb2-44df-ba4a-8c2fd26496f4">

**Comment by sschmidTU** [comment]
Oh, right, we have `findVisibleGraphicalMeasure()`. That used to not exist, in fact I wrote it, haha. Less lines of code to use that here, yes.

```diff
diff --git a/src/OpenSheetMusicDisplay/Cursor.ts b/src/OpenSheetMusicDisplay/Cursor.ts
index 14f969e4..ee990750 100644
--- a/src/OpenSheetMusicDisplay/Cursor.ts
+++ b/src/OpenSheetMusicDisplay/Cursor.ts
@@ -216,7 +216,11 @@ export class Cursor {
     height = endY - y;
 
     // Update the graphical cursor
-    const measurePositionAndShape: BoundingBox = this.graphic.findGraphicalMeasure(currentMeasureIndex, 0).PositionAndShape;
+    const visibleMeasure: GraphicalMeasure = this.findVisibleGraphicalMeasure(currentMeasureIndex);
+    if (!visibleMeasure) {
+      return;
+    }
+    const measurePositionAndShape: BoundingBox = visibleMeasure.PositionAndShape;
     this.updateWidthAndStyle(measurePositionAndShape, x, y, height);
 
     if (this.openSheetMusicDisplay.FollowCursor && this.cursorOptions.follow) {
```

**Comment by sschmidTU** [comment]
I think it depends on the sample whether cursor type 3 highlighted anything at all beforehand. But in any case, it should be fixed now, see PR #1429.

**Comment by sschmidTU** [comment]
Should be fixed by #1429.
I'll release OSMD 1.8.1 including the fix shortly.
If it doesn't work for your samples, we can still adjust the fix afterwards :)

**Comment by boblete** [comment]
All fixed my end
